src/datasources/asset/AssetDataSource.ts

```diff
@@ -0,0 +1,325 @@
+      let dataWithoutOverlaps: Array<Interval<number>> = [];
+      if (data.length) {
+        const extractedTimestamps = extractTimestampsFromData(data);
+        // Fill missing endTimestamps
+        const patchedData = patchMissingEndTimestamps(extractedTimestamps);
+        // Removes data outside the grafana 'from' and 'to' range from an array
+        const filteredData = filterDataByTimeRange(patchedData, from, to);
+        // Merge overlapping utilizations
+        dataWithoutOverlaps = mergeOverlappingIntervals(filteredData);
+      }
+      // Divide given time range to peak/non-peak intervals
+      const businessIntervals = divideTimeRangeToBusinessIntervals(new Date(from), new Date(to), workingHoursPolicy, peakDays, timeFrequency);
+      // Group raw utilization data by intervals
+      const overlaps = groupDataByIntervals(businessIntervals, dataWithoutOverlaps, peakDays, isPeak);
+      // Calculate utilization percentage by intervals
+      const utilization = calculateUtilization(overlaps);
+      // patch zero points to data for better visualisation
+      const patchedUtilization = patchZeroPoints(utilization, new Date(from), new Date(to), timeFrequency);
+
+      let utilization_data: { id: string, datetimes: number[], values: number[] } =
+        {
+          id: arrayOfEntityIds[index],
+          datetimes: patchedUtilization.map(v => dateTime(v.day).valueOf()),
+          values: patchedUtilization.map(v => v.utilization)
+        }
+      resultArray.push(utilization_data);
+    }
+
+    return resultArray;
+  }
+
+  async getServicePolicy(): Promise<ServicePolicyModel> {
+    return await this.get<ServicePolicyModel>(this.baseUrl + '/policy');
+  }
+
+  async queryAssets(filter = '', take = -1): Promise<AssetModel[]> {
+    let data = { filter, take };
+    try {
+      let response = await this.post<AssetsResponse>(this.baseUrl + '/query-assets', data);
+      return response.assets;
+    } catch (error) {
+      throw new Error(`An error occurred while querying assets: ${error}`);
+    }
+  }
+
+  async querySystems(filter = '', projection = defaultProjection): Promise<SystemMetadata[]> {
+    try {
+      let response = await this.post<{ data: SystemMetadata[] }>(this.sysmgmtUrl + '/query-systems', {
+        filter: filter,
+        projection: `new(${projection.join()})`,
+        orderBy: defaultOrderBy,
+      });
+      return response.data;
+    } catch (error) {
+      throw new Error(`An error occurred while querying systems: ${error}`);
+    }
+  }
+
+  async testDatasource(): Promise<TestDataSourceResponse> {
+    await this.get(this.baseUrl + '/assets');
+    return { status: 'success', message: 'Data source connected and authentication successful!' };
+  }
+
+  async metricFindQuery({ minionId }: AssetVariableQuery): Promise<MetricFindValue[]> {
+    const minionID = this.templateSrv.replace(minionId);
+    const filter = minionID ? `Location.MinionId = \"${minionID}\"` : '';
```

Comment:
```suggestion
    const filter = minionID ? `Location.MinionId = "${minionID}"` : '';
```

I don't think escaping the slashes is necessary in a template literal.